zproject/urls.py

```diff
@@ -202,6 +202,8 @@
         {'GET': 'zerver.views.messages.get_message_edit_history'}),
     url(r'^messages/matches_narrow$', rest_dispatch,
         {'GET': 'zerver.views.messages.messages_in_narrow_backend'}),
+    url(r'^messages/markdown$', rest_dispatch,
+        {'GET': 'zerver.views.messages.json_fetch_raw_list_messages'}),
```

Comment:
Why do we need a new API endpoint?  We should be able to do everything we want with using `GET /messages` to fetch the messages (passing `apply_markdown=False` to get the raw content), and then having the client do the little bit of formatting of sender names, date, etc.

Reply:
Hi Tim, the main reason is the performance. For example, if you need the data of a message, it wouldn't important. But, if you have a range of 10, 20 or >30  messages, it would be the same number of requests to the server. In the frontend, it would be slow too.

Reply:
@timabbott, if you are telling me about:
^messages$ -> 'GET': 'zerver.views.messages.get_messages_backend'
It works with a homogeneus range, but we can have a list with individual messages mixed with little ranges too.

Reply:
Should we add a new option to the existing endpoint to allow it to accept individual ranges & message ids instead of adding a new endpoint?

Reply:
I'm a little puzzled here -- Is that idea that one will be able to manually select/deselect individual messages in the UI?  If it's a range of things within a view that the Zulip webapp can display, there's a `get_messages` query that will return that range of messages.

I would potentially be open to add a "Get messages by ID" option to `get_messages` where you just provide a list of message IDs and no anchor or anything, but I'm not sure it's relevant to what you're trying to do here.

Reply:
The feature requires something like the second option ("Get messages by ID"), because we are sending to the endpoint an message ID array.
If this is possible:
1. The view is "get_messages_backend", isn't it?

`def get_messages_backend(request: HttpRequest, user_profile: UserProfile,
                         anchor_val: Optional[str]=REQ(
                             'anchor', str_validator=check_string, default=None),
                         num_before: int=REQ(converter=to_non_negative_int),
                         num_after: int=REQ(converter=to_non_negative_int),
                         narrow: OptionalNarrowListT=REQ('narrow', converter=narrow_parameter, default=None),
                         use_first_unread_anchor_val: bool=REQ('use_first_unread_anchor',
                                                               validator=check_bool, default=False),
                         client_gravatar: bool=REQ(validator=check_bool, default=False),
                         apply_markdown: bool=REQ(validator=check_bool, default=True)) -> HttpResponse`

2. num_before and num_after would have a default value?
3. should we add a boolean param "get_by_ids" and a new param "message_ids"?

Reply:
@afreydev I'm still not clear on why the frontend here needs "Get messages by ID" -- can you answer that question clearly?

Reply:
For this feature, the user can select "continuous or discontinuous" group of messages. If you check the gif, sometimes you can select 1 or 2 or 3 consecutive messages but in other cases select multiple not consecutive messages. 
I have updated the version, in this I have deleted the new endpoint but I have added the param message_id_list to get_messages_backend. Couldn't it work like that?

Reply:
![copy-paste4](https://user-images.githubusercontent.com/43050896/75832613-9bd95800-5d84-11ea-9001-b06845471ed8.gif)